src/global.c

```diff
@@ -201,6 +198,17 @@ int git_libgit2_shutdown(void)
 #endif
 	}
 
+	/*
+	 * Always clean up thread-specific global data before
+	 * free'ing the TLS index. We obviously need to do this
+	 * before calling `TlsFree()`, but in fact after
+	 * shutting down all subsystems.
+	 */
+	git__free_tls_data();
```

Comment:
Mh, in fact this isn't quite right either, as we'll now always free TLS data when calling `git_libgit2_shutdown()`. But if the same thread has called `git_libgit2_init()` multiple times, then we definitely don't want that. We could fix the issue by adding another reference count to the global state, but I dunno...